src/launcher.c

```diff
@@ -293,10 +315,28 @@ static void *load_app(char *name)
     }
   }
 
-  if (patch_svc(code, size) != 0) {
+  uint32_t start = 0;
+  uint32_t code_length = size;
+  if (app->elf.code_start && app->elf.code_end) {
+    start = app->elf.code_start - TEXT_START;
+    code_length = app->elf.code_end - app->elf.code_start;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_WRITE) != 0) {
+    warn("could not update mprotect in rw mode for app");
     goto error;
   }
 
+  if (patch_svc(code + start, code_length) != 0) {
+    warn("could not patch svcs in app code");
+    goto error;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_EXEC) != 0) {
+    warn("could not update mprotect in rx mode for app");
+    _exit(1);
```

Comment:
Why an `exit` here, and a `goto error` above ?

Reply:
fixed in a later commit sorry